### PR TITLE
DNN-6567: GetFoldersByPermissions returns wrong records

### PR DESCRIPTION
**Pull request by sleupold**

### CODE REVIEW

File: Website/Providers/DataProviders/SqlDataProvider/08.01.00.01.SqlDataProvider [hunk 1]
```diff
@@ -166,3 +166,75 @@ AS
...
+
+    IF (IsNull(@PortalID, -1) = -1 AND @IsSuperUser != 0) OR (@IsAdmin != 0)
+        -- shortcut for admins/superusers: return all folders (they are having always access to all of them)
+        SELECT F.*
+         FROM  {databaseOwner}[{objectQualifier}Folders] F
+         WHERE IsNull(F.PortalID, -1) = IsNull(@PortalID, -1)
+           AND (F.FolderID   = @FolderID   OR IsNull(@FolderID,    -1) =  -1)
+           AND (F.FolderPath = @FolderPath OR IsNull(@FolderPath, N'') = N'')
+         ORDER BY F.PortalID, F.FolderPath
+         OPTION (OPTIMIZE FOR (@PortalID UNKNOWN));
```
Comment (IDisposable) [inline Website/Providers/DataProviders/SqlDataProvider/08.01.00.01.SqlDataProvider]: Better
```
 OPTION (OPTIMIZE FOR (@PortalID UNKNOWN,  @FolderID UNKNOWN, @FolderPath UNKNOWN ));
```

File: Website/Providers/DataProviders/SqlDataProvider/08.01.00.01.SqlDataProvider [hunk 2]
```diff
@@ -166,3 +166,75 @@ AS
...
+                 WHERE (UserID = @UserID OR (RoleID Is Not Null AND {databaseOwner}[{objectQualifier}UserIsInRole](@UserID, RoleID) = 1))
+                   AND PermissionID IN (@AddPerm, @ReadPerm, @BrowsePerm, @AllPerm)
+                   AND AllowAccess = 1)
+           AND NOT FolderId IN -- denied folders
+                (SELECT DISTINCT FolderId FROM {databaseOwner}[{objectQualifier}FolderPermission]
+                 WHERE (UserID = @UserID OR (RoleID Is Not Null AND {databaseOwner}[{objectQualifier}UserIsInRole](@UserID, RoleID) = 1))
+                   AND PermissionID IN (@AddPerm, @ReadPerm, @BrowsePerm, @WritePerm)
+                   AND AllowAccess = 0)
+         ORDER BY PortalID, FolderPath
+         OPTION (OPTIMIZE FOR (@FolderPath Unknown, @FolderId Unknown, @UserID Unknown));
```
Comment (IDisposable) [inline Website/Providers/DataProviders/SqlDataProvider/08.01.00.01.SqlDataProvider]: Missing `@PortalId` from that list...

```
 OPTION (OPTIMIZE FOR (@PortalID UNKNOWN, @FolderID UNKNOWN, @FolderPath UNKNOWN. @UserID UNKNOWN));
```

File: Website/Providers/DataProviders/SqlDataProvider/08.01.00.01.SqlDataProvider (hunk 1 above)
Comment (sleupold) [inline Website/Providers/DataProviders/SqlDataProvider/08.01.00.01.SqlDataProvider]: thanks, fixed

**Comment by sleupold** [comment]
I will resubmit asap

**Comment by mitchelsellers** [comment]
Closing, @sleupold will be submitting a new pull request

**Pull request closed without merging**